### PR TITLE
Fix refcounting/deallocation mechanism for stack allocated objects

### PR DESCRIPTION
**Pull request by neithernut**
This is done by
- introducing a flag denoting ojects allocated on the heap
- Setting that flag in each existing derived "class".

### CODE REVIEW

File: src/objects/object.c [hunk 1]
```diff
@@ -138,32 +138,35 @@ struct ws_object*
 ws_object_getref(
     struct ws_object* self
 ) {
-    if (self) {
-        ws_object_lock_write(self);
-        self->ref_counting.refcnt++;
-        ws_object_unlock(self);
+    if (!self || (self->settings & WS_OBJECT_HEAPALLOCED)) {
```
Comment (matthiasbeyer) [inline src/objects/object.c]: `not` missing.

File: src/objects/object.c [hunk 2]
```diff
@@ -138,32 +138,35 @@ struct ws_object*
...
 }
 
 void
 ws_object_unref(
     struct ws_object* self
 ) {
-    if (self) {
-        ws_object_lock_write(self);
-        self->ref_counting.refcnt--;
+    if (!self || (self->settings & WS_OBJECT_HEAPALLOCED)) {
```
Comment (matthiasbeyer) [inline src/objects/object.c]: `not` missing.

**Comment by matthiasbeyer** [comment]
Go for squash.

**Comment by neithernut** [comment]
Done.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**